app/shared/src/commonMain/kotlin/ui/subject/components/comment/EditComment.kt

```diff
@@ -0,0 +1,635 @@
+            focusedBorderColor = Color.Transparent,
+            cursorColor = MaterialTheme.colorScheme.primary,
+        ),
+        placeholder: @Composable (() -> Unit)? = { Text(hint ?: "") }
+    ) {
+        OutlinedTextField(
+            value = value,
+            onValueChange = onValueChange,
+            modifier = modifier,
+            textStyle = MaterialTheme.typography.bodyMedium.merge(
+                fontSize = 15.5.sp,
+                color = LocalContentColor.current.looming(),
+            ),
+            shape = shape,
+            enabled = enabled,
+            maxLines = maxLines,
+            placeholder = placeholder,
+            colors = colors,
+            interactionSource = interactionSource,
+        )
+    }
+
+    @Composable
+    fun Preview(
+        content: UIRichText,
+        modifier: Modifier = Modifier,
+        contentPadding: PaddingValues = PaddingValues(0.dp)
+    ) {
+        Surface(
+            modifier = modifier,
+            shape = RoundedCornerShape(12.dp),
+            color = MaterialTheme.colorScheme.surface,
+        ) {
+            LazyColumn {
+                item {
+                    RichText(
+                        elements = content.elements,
+                        modifier = Modifier.padding(contentPadding),
+                    )
+                }
+            }
+        }
+    }
+
+    @Composable
+    fun ActionButton(
+        imageVector: ImageVector,
+        onClick: () -> Unit,
+        modifier: Modifier = Modifier,
+        enabled: Boolean = true,
+        iconSize: Dp = 22.dp,
+        indication: Indication? = rememberRipple(
+            bounded = false,
+            radius = 20.dp, /* IconButtonTokens.StateLayerSize / 2 */
+        )
+    ) {
+        me.him188.ani.app.ui.foundation.IconButton(
+            modifier = modifier,
+            enabled = enabled,
+            onClick = onClick,
+            indication = indication,
+        ) {
+            Icon(
+                imageVector = imageVector,
+                contentDescription = null,
+                modifier = Modifier.size(iconSize),
+            )
+        }
+    }
+
+    @Composable
+    fun ActionRow(
+        sendTarget: CommentSendTarget?,
+        onClickBold: () -> Unit,
+        onClickItalic: () -> Unit,
+        onClickUnderlined: () -> Unit,
+        onClickStrikethrough: () -> Unit,
+        onClickMask: () -> Unit,
+        onClickImage: () -> Unit,
+        onClickUrl: () -> Unit,
+        onClickEmoji: () -> Unit,
+        onSend: () -> Unit,
+        onPreview: () -> Unit,
+        modifier: Modifier = Modifier,
+        previewing: Boolean = false,
+        sending: Boolean = false,
+    ) {
+        // maybe we can extract the layout
+        val size = ActionButtonSize.dp
+        var actionExpanded by rememberSaveable { mutableStateOf(false) }
+        val collapsedActionAnim by animateDpAsState(if (actionExpanded) size else 0.dp)
```

Comment:
```suggestion
        val expandableActionWidth by animateDpAsState(if (actionExpanded) size else 0.dp)
```
突出这个东西的用途是给那些能够收起来的 action, 要比突出这是个动画属性更重要